contracts/meta/MetaSwapDeposit.sol

```diff
@@ -0,0 +1,529 @@
+     * @param tokenIndex the index of the token you want to receive
+     * @param minAmount the minimum amount to withdraw, otherwise revert
+     * @param deadline latest timestamp to accept this transaction
+     * @return amount of chosen token user received
+     */
+    function removeLiquidityOneToken(
+        uint256 tokenAmount,
+        uint8 tokenIndex,
+        uint256 minAmount,
+        uint256 deadline
+    ) external returns (uint256) {
+        uint8 baseLPTokenIndex = uint8(metaTokens.length - 1);
+        uint8 baseTokensLength = uint8(baseTokens.length);
+
+        metaLPToken.safeTransferFrom(msg.sender, address(this), tokenAmount);
+
+        IERC20 token;
+        if (tokenIndex < baseLPTokenIndex) {
+            metaSwap.removeLiquidityOneToken(
+                tokenAmount,
+                tokenIndex,
+                minAmount,
+                deadline
+            );
+            token = metaTokens[tokenIndex];
+        } else if (tokenIndex < baseLPTokenIndex + baseTokensLength) {
+            uint256 removedBaseLPTokenAmount =
+                metaSwap.removeLiquidityOneToken(
+                    tokenAmount,
+                    baseLPTokenIndex,
+                    0,
+                    deadline
+                );
+
+            baseSwap.removeLiquidityOneToken(
+                removedBaseLPTokenAmount,
+                tokenIndex - baseLPTokenIndex,
+                minAmount,
+                deadline
+            );
+            token = baseTokens[tokenIndex - baseLPTokenIndex];
+        } else {
+            revert("out of range");
+        }
+
+        uint256 amountWithdrawn = token.balanceOf(address(this));
+        token.safeTransfer(msg.sender, amountWithdrawn);
+        return amountWithdrawn;
+    }
+
+    /**
+     * @notice Remove liquidity from the pool, weighted differently than the
+     * pool's current balances. Withdraw fee that decays linearly
+     * over period of 4 weeks since last deposit will apply.
+     * @param amounts how much of each token to withdraw
+     * @param maxBurnAmount the max LP token provider is willing to pay to
+     * remove liquidity. Useful as a front-running mitigation.
+     * @param deadline latest timestamp to accept this transaction
+     * @return amount of LP tokens burned
+     */
+    function removeLiquidityImbalance(
+        uint256[] calldata amounts,
+        uint256 maxBurnAmount,
+        uint256 deadline
+    ) external returns (uint256) {
+        IERC20[] memory memBaseTokens = baseTokens;
+        IERC20[] memory memMetaTokens = metaTokens;
+        uint256[] memory metaAmounts = new uint256[](memMetaTokens.length);
+        uint256[] memory baseAmounts = new uint256[](memBaseTokens.length);
+
+        {
+            uint256 numOfAllTokens =
+                memBaseTokens.length + memMetaTokens.length - 1;
+            require(amounts.length == numOfAllTokens, "out of range");
+        }
```

Comment:
j/w is the performance of this the same as not assigning `numOfAllTokens` and directly using `memBaseTokens.length + memMetaTokens.length - 1` in the require?

Reply:
The difference is minimal (better to use directly) but there is no need for storing `numOfAllTokens` in this function. Good catch!